domain/keymanager/state/state.go

```diff
@@ -33,196 +35,415 @@ func NewState(factory database.TxnRunnerFactory) *State {
 	}
 }
 
-// AddPublicKeyForUser is responsible for adding one or more ssh public keys for
-// a user. If no keys are provided to the operation then nothing will happen and
-// no error produced. The following errors can be expected.
-// [keyerrors.PublicKeyAlreadExists] - When one of the public keys being added
-// for a user already exists.
-func (s *State) AddPublicKeysForUser(
+// checkModelExists is responsible for checking that a given model exists within
+// the controller. If the model exists no error is returned otherwise an error
+// satisfying [modelerrors.NotFound] is returned.
+func (s *State) checkModelExists(
 	ctx context.Context,
-	userId user.UUID,
-	publicKeys []keymanager.PublicKey,
+	modelUUID model.UUID,
+	tx *sqlair.TX,
 ) error {
-	if len(publicKeys) == 0 {
-		return nil
+	modelUUIDVal := modelUUIDValue{UUID: modelUUID.String()}
+
+	modelExistsStmt, err := s.Prepare(`
+SELECT (uuid) AS (&modelUUIDValue.model_uuid)
+FROM v_model
+WHERE uuid = $modelUUIDValue.model_uuid
+`, modelUUIDVal)
+	if err != nil {
+		return errors.Errorf(
+			"creating model exists statement when checking if model %q exists: %w",
+			modelUUID, err,
+		)
 	}
 
-	db, err := s.DB()
+	err = tx.Query(ctx, modelExistsStmt, modelUUIDVal).Get(&modelUUIDVal)
+	if errors.Is(err, sqlair.ErrNoRows) {
+		return errors.Errorf(
+			"checking model %q exists, model not found", modelUUID,
+		).Add(modelerrors.NotFound)
+	} else if err != nil {
+		return errors.Errorf(
+			"checking model %q exists: %w", modelUUID, err,
+		)
+	}
+
+	return nil
+}
+
+// checkUserExists is responsible for checking that a given user exists with the
```

Comment:
".. user exists and is active .. " ?

Reply:
no active is a different meaning.  active means they have authn. We don't consider authn in this context.